docs/transaction.md

```diff
@@ -0,0 +1,261 @@
+# Life of a transaction
+
+This document describes how Dragonfly transactions provide atomicity and serializability for its multi-key and multi-command operations.
+
+## Definitions
+
+### Serializability
+
+Serializability is an isolation level for database transactions. Serializability describes multiple transactions, where a transaction is usually composed of multiple operations on multiple objects.
+
+Database can executed transactions in parallel (and the operations in parallel). Serializability guarantees the result is the same with, as if the transactions were executed one by one. i.e. to behave like executed in a serial order.
+
+Serializability doesn’t guarantee the resulting serial order respects recency. I.e. the serial order can be different from the order in which transactions were actually executed. E.g. Tx1 begins earlier than Tx2, but the result behaves as if Tx2 executed before Tx1. That is also to say, to satisfy the same Serializability, there can be more than one possible execution schedulings.
+
+### Linearizability
+
+Linearizability (also known as atomic consistency, strong consistency, immediate consistency) describes reads and writes on a single object (stores a single value). it doesn’t involve multiple objects. It doesn’t involve “transaction”, which groups multiple objects. It treats each operation as an atom, i.e. to take effect in a single time point, rather than a timespan.
+
+Note that simple, single-key operations in Dragonfly are already linearizable. Indeed, in a shared-nothing architecture each shard-thread performs operations on its keys sequentially.
```

Comment:
In a shared nothing architecture each shard-thread performs operations on its keys sequentially, therefor single-key operations in DragonflyDB are already linearizable and serializability. The compexity...

Move this line to the top where you write whats in this doc

Reply:
I think that also multi key operations ensure Linearizability in the shared nothing architecture. What we handle here is just Serializability

Reply:
From the definistion you added on serializable and linearizable

Linearizability (also known as atomic consistency, strong consistency, immediate consistency) describes reads and writes on a single object (stores a single value). it doesn’t involve multiple objects.
From what I understand here it has nothing to do with the multi key command

Database can executed transactions in parallel (and the operations in parallel). Serializability guarantees the result is the same with, as if the transactions were executed one by one. i.e. to behave like executed in a serial order.
From what I understand here this exactly answers the issue that we can not return "x=a2, y=b1"

Reply:
review again this comment. You describe Linearizability (for single object) and not atomicity.
Above you write 
"This document describes how Dragonfly transactions provide atomicity and serializability for its multi-key and multi-command operations."
here you describe Linearizability and below you write 
The complexity rises when we need to provide strict-serializability (aka serializability and linearizability) for multiple keys.
The terms are confusing.
Its better if you use one term, stick with it and describe it. f.e if you want to decribe how  Dragonfly transactions provide atomicity and serializability, than define atomicity and serializability. The Linearizability definition is redundant